packages/core/src/app/customer/customWalletButton/AmazonPayV2Button.tsx

```diff
@@ -5,7 +5,7 @@ import CheckoutButton, { CheckoutButtonProps } from '../CheckoutButton';
 const AmazonPayV2Button: FunctionComponent<CheckoutButtonProps> = (props) => {
     useEffect(() => {
         beautifyAmazonButton();
-    }, []);
+    }, [props]);
```

Comment:
Do we need to pass `props` here? Don't think it's used in the effect?

Reply:
I think yes because container width which we depend on here is set later and wen trying to use 
```
useEffect(() => {
        beautifyAmazonButton();
    }, []);
```
the propper width is not set yet.

Reply:
The interface of props does not seem to contain the width.
```
export interface CheckoutButtonProps {
    containerId: string;
    methodId: string;
    isShowingWalletButtonsOnTop?: boolean;
    deinitialize(options: CustomerRequestOptions): void;
    initialize(options: CustomerInitializeOptions): void;
    onError?(error: Error): void;
}
```
Could you please specify which prop it depends on?

Reply:
Actually It doesn’t depend on any specific information in props. It rather depends on that AmazonPayV2Button component is receiving new props object, which cause the `beautifyAmazonButton()` function call.
Let me explain:
Previously `beautifyAmazonButton()` was called after first render:

```
useEffect(() => {
        beautifyAmazonButton();
    }, []);
```

But it was to early to set container width to 100% as in this moment its parent div has not yet set proper width. In next steps AmazonPayV2Button component is receiving new props object (in mean time parent div is getting proper width) and then when `beautifyAmazonButton()` function is called and the container is getting proper 100% width.

Reply:
Can I know how many times will this `useEffect` be called in a normal checkout page load after passing `prop` ?

Reply:
I checked and as I see this useEffect is called 13 times

Reply:
I think we would all agree 13 is a big number.
It's not a blocker, but could you please try use just few props to reduce the rendering times?

Reply:
Is this 13, with adding the props object? If yes then I align with Peng, maybe we need to rethink how the width is set as this feels like a hack and can cause some side effects.